File: backend/entity/student_post.go
package entity

import (
	"gorm.io/gorm"
)

// StudentPost เก็บข้อมูลโพสต์สำหรับประกาศหางานของนักศึกษา
type StudentPost struct {
	gorm.Model

	// --- ข้อมูลเฉพาะของโพสต์ ---
	Title                string `json:"title" gorm:"type:varchar(255);not null"`
	
	Availability         string `json:"availability" gorm:"type:varchar(255);not null"`
	PreferredLocation    string `json:"preferred_location" gorm:"type:varchar(255);not null"`
	ExpectedCompensation string `json:"expected_compensation" gorm:"type:text"`
	Introduction         string `json:"introduction" gorm:"type:text"`
	PortfolioURL         string `json:"portfolio_url" gorm:"type:varchar(500)"`
	Status               string `json:"status" gorm:"type:varchar(50);default:'active'"`
	Skills []*Skill `gorm:"many2many:student_post_skills;" json:"skills"`

	EmploymentTypeID *uint `json:"employment_type_id"`
	EmploymentType   *EmploymentType `json:"employment_type" gorm:"foreignKey:EmploymentTypeID"`

	// --- Foreign Key & Relations ---

	// 1. ความสัมพันธ์กับ Student (เจ้าของโพสต์)
	StudentID *uint   `json:"student_id" gorm:"not null"`
	Student   Student `json:"student" gorm:"foreignKey:StudentID;references:ID"`

	// 2. ความสัมพันธ์กับไฟล์แนบ
	Attachments []StudentPostAttachment `json:"attachments" gorm:"foreignKey:StudentPostID"`



	
}

// กำหนดชื่อตารางให้ชัดเจน
func (StudentPost) TableName() string {
	return "student_posts"
}

File: frontend/src/interfaces/studentpost.ts
// src/interfaces/studentpost.ts

import type { Student } from "./student";
import type { Skill } from "./skill";
import type { EmploymentType } from "./employment_type";


// Interface สำหรับไฟล์แนบ (ใช้แทน Attachment เดิม)
export interface StudentPostAttachment {
  ID?: number; // Optional ID
  url: string;
  name: string;
  type: string;
  student_post_id?: number; // Optional
}

// Interface สำหรับ Props ของ Modal โดยเฉพาะ
export interface CreateStudentPostModalProps {
  visible: boolean;
  onClose: () => void;
  onSuccess: () => void;
}

export interface StudentPostAttachment {
    ID?: number;
    url: string;
    name: string;
    type: string;
    student_post_id?: number;
  }
  
// Interface หลักสำหรับโพสต์ของนักศึกษา (คงเดิม)
export interface StudentPost {
  ID: number;
  CreatedAt: string;
  UpdatedAt: string;
  
  // --- ข้อมูลหลักของโพสต์ ---
  title: string;
  job_type: string;
  availability: string;
  preferred_location: string;
  expected_compensation?: string;
  introduction?: string;
  portfolio_url?: string;
  status: string;
  skills: Skill[];
  profile_image_url?: string;

  // --- ข้อมูลความสัมพันธ์ (Relations) ---
  student_id?: number;
  student?: Student;

  attachments?: StudentPostAttachment[];
  employment_type?: EmploymentType; 
}
export interface EditStudentPostModalProps {
    visible: boolean;
    onClose: () => void;
    onSuccess: () => void;
    post: StudentPost | null; // รับ post ที่จะแก้ไข
  }